Agents/ForecastingAgent/forecastingagent/agent/forecasting.py

```diff
@@ -0,0 +1,338 @@
+                raise TypeError(f"Derivation {derivationIRI}: More than one 'om:Quantity' IRI provided to forecast.")
+        elif inputs.get(OWL_THING):
+            msg = f"Derivation {derivationIRI}: No 'om:Quantity' IRI provided to forecast. "
+            msg += "Trying to retrieve 'owl:Thing' IRI to forecast."
+            self.logger.warning(msg)
+            # Extract unique owl:Thing IRI
+            inp = _find_unique_owl_thing_iris(inputs)
+            # Check whether only one input has been provided
+            if len(inp) == 1:
+                input_iris['iri_to_forecast'] = inp[0]
+            elif len(inp) == 0:
+                self.logger.error(f"Derivation {derivationIRI}: Neither 'om:Quantity' nor 'owl:Thing' IRI provided to forecast.")
+                raise TypeError(f"Derivation {derivationIRI}: Neither 'om:Quantity' nor 'owl:Thing' IRI provided to forecast.")
+            else:
+                self.logger.error(f"Derivation {derivationIRI}: No unique 'owl:Thing' IRI provided to forecast.")
+                raise TypeError(f"Derivation {derivationIRI}: No unique 'owl:Thing' IRI provided to forecast.")
+        
+        # 2) For existing derivations
+        else:
+            # If neither om:Quantity nor owl:Thing is marked up directly as input
+            # (e.g., in cases where a subclass of om:Quantity is forecasted), try to
+            # extract the appropriate IRI
+            type_to_forecast = [k for k in inputs if k not in input_iris.keys()]
+            if len(type_to_forecast) != 1:
+                self.logger.error(f"Derivation {derivationIRI}: Rdf:type to forecast could not be determined.")
+                raise TypeError(f"Derivation {derivationIRI}: Rdf:type to forecast could not be determined.")
+            else:
+                inp = inputs.get(type_to_forecast[0])
+                # Check whether only one input has been provided
+                if len(inp) == 1:
+                    input_iris['iri_to_forecast'] = inp[0]
+                else:
+                    self.logger.error(f"Derivation {derivationIRI}: IRI to forecast could not be determined.")
+                    raise TypeError(f"Derivation {derivationIRI}: IRI to forecast could not be determined.")
+
+        return input_iris
+
+
+    def process_request_parameters(self, derivation_inputs: DerivationInputs, 
+                                   derivation_outputs: DerivationOutputs):
+        """
+        This method takes 
+            1 IRI of OntoTimeSeries:ForecastingModel (model to be used for forecasting)
+            1 IRI of OntoTimeSeries:Frequency (frequency of forecasted time series)
+            1 IRI of Time:TimeInterval (time interval to be forecasted)
+            1 IRI of Time:TimeDuration (data length prior to forecast interval
+                                        to be used for training/data scaling)
+            1 IRI of OM:Quantity or owl:Thing (the concept associated with the 
```

Comment:
I still think it's less cumbersome to define `om:Quantity` as the domain of `OntoTimeSeries:hasTimeSeries` as opposed to `owl:Thing` and it will work for most of the use cases. This might be a question to @sm453.

Reply:
Well, `om:Quantity` is much more restrictive than `owl:Thing`. Not everyone uses OM. Unless it causes _substantial_ trouble to use `owl:Thing`, I'd go for that. Having said that, if the agent can _only_ forecast our own time series class, and that is restricted to a certain type, then there is no benefit in being more general here. Another question: You say "most" of the use cases. Do we already have use cases where `om:Quantity` is not general enough?

Reply:
The thing is (pardon the pun), as long as the derived information framework matches inputs based on `rdf:type`, it makes no sense to use `owl:Thing` as an input type, because _everything_ matches that (_any_ `owl:Class` is a subclass of `owl:Thing`!). For that reason, we must not use `owl:Thing` here (at least until issue #896 is resolved).

So, I would suggest sticking to `om:Quantity` for the time being. Let's hope that no-one is using time series of e.g. OntoCAPE quantities...

Reply:
It does not cause *substantial* trouble to use `owl:Thing`:
1) Since only one time series is forecasted (and all other derivation input types are clearly defined), it is easily possible to extract the IRI to forecast unambiguously
2) The agent does not take our own time series concept instances directly, but rather the associated dataIRIs to derive the forecasts; hence, the time series design does not restrict the possible classes to forecast
3) For energy demand modelling, one might potentially want to forecast (room) occupancy, or the availability of boilers for the district heating. So, there are cases where target concepts are not defined as `om:Quantity`. Furthermore, `owl:Thing` is more in line with the time series ontology

Reply:
What about defining "(room) occupancy" and "availability of boilers for the district heating" as subclasses of `om:Quantity`?